centreon/src/Core/HostTemplate/Application/UseCase/PartialUpdateHostTemplate/PartialUpdateHostTemplate.php

```diff
@@ -185,103 +187,42 @@ private function updateMacros(PartialUpdateHostTemplateRequest $request, HostTem
-            if (isset($updatedMacros[$macro->getName()])) {
-                $macro->setOrder($order);
-                $this->writeHostMacroRepository->update($macro);
-                ++$order;
-
-                continue;
-            }
-            if (isset($addedMacros[$macro->getName()])) {
-                $macro->setOrder($order);
-                if ($macro->getDescription() === '') {
-                    $macro->setDescription(
-                        isset($commandMacros[$macro->getName()])
-                        ? $commandMacros[$macro->getName()]->getDescription()
-                        : ''
-                    );
-                }
-                $this->writeHostMacroRepository->add($macro);
-                ++$order;
-
-                continue;
-            }
-            if (isset($unchangedMacros[$macro->getName()])) {
-                if (
-                    isset($directMacros[$macro->getName()])
-                    && $directMacros[$macro->getName()]->getOrder() !== $order
-                ) {
-                    // macro is the same but its order has changed
-                    $macro->setOrder($order);
-                    $this->writeHostMacroRepository->update($macro);
-                }
-                ++$order;
-            }
+        foreach ($macrosDiff['updated'] as $macro) {
+            $this->writeHostMacroRepository->update($macro);
         }
-    }
 
-    /**
-     * Create macros object from the request dto.
-     * Retrieve original value if macro is of type password.
-     *
-     * @param array<array{name:string,value:string|null,is_password:bool,description:string|null}> $requestedMacros
-     * @param int $hostTemplateId
-     * @param array<string,HostMacro> $directMacros
-     * @param array<string,HostMacro> $inheritedMacros
-     *
-     * @throws \Throwable
-     *
-     * @return array<string,HostMacro>
-     */
-    private function createMacros(
-        array $requestedMacros,
-        int $hostTemplateId,
-        array $directMacros,
-        array $inheritedMacros
-    ): array {
-        $macros = [];
-        foreach ($requestedMacros as $data) {
-            $macroName = mb_strtoupper($data['name']);
-            // Note: do not handle vault at the moment
-            $macroValue = $data['value'] ?? '';
-            if ($data['is_password'] && $data['value'] === null) {
-                // retrieve actual password value
-                $macroValue = isset($directMacros[$macroName])
-                    ? $directMacros[$macroName]->getValue()
-                    : (isset($inheritedMacros[$macroName]) ? $inheritedMacros[$macroName]->getValue() : '');
+        foreach ($macrosDiff['added'] as $macro) {
+            if ($macro->getDescription() === '') {
+                $macro->setDescription(
+                    isset($commandMacros[$macro->getName()])
+                    ? $commandMacros[$macro->getName()]->getDescription()
+                    : ''
+                );
```

Comment:
Subjective readibility of 
```suggestion
                $commandMacro = $commandMacros[$macro->getName()] ?? null;
                $macro->setDescription(
                    $commandMacro ? $commandMacro->getDescription() : ''
                );
```